client/rpc.go

```diff
@@ -253,9 +263,40 @@ func (c RPC) QueryRequestFailureReason(id uint64) (string, error) {
 	return "", fmt.Errorf("no reason found")
 }
 
-func (c RPC) GetSignature(_ uint64) ([]byte, error) {
-	// TODO: Implement when need TSS signature
-	return []byte{}, nil
+func (c RPC) GetSignature(signingID uint64) (*SigningResult, error) {
+	resultCh := make(chan *SigningResult, len(c.nodes))
+	failCh := make(chan struct{}, len(c.nodes))
+
+	for _, node := range c.nodes {
+		go func(node *rpchttp.HTTP) {
+			c.logger.Debug("GetSignature", "Try to get signature from %s", node.Remote())
+
+			res, err := getSigningResult(c.ctx.WithClient(node), signingID)
+			if err != nil {
+				c.logger.Warning(
+					"GetSignature",
+					"Fail to get signature from %s with error %s",
+					node.Remote(), err,
+				)
+				failCh <- struct{}{}
+				return
+			}
+
+			signingResult := convertSigningResponse(res)
```

Comment:
do we want to consider whether an empty signingResult is considered valid?

Reply:
likely to happen ah, at least should have currentGroupSigning result or get an error. Anyway, I added a condition to check that hai la na.